example/advection/advection.cpp

```diff
@@ -0,0 +1,399 @@
+  int ie = pmb->ie; int je = pmb->je; int ke = pmb->ke;
+
+  Real min_dt = std::numeric_limits<Real>::max();
+  ParArrayND<Real> dx0("dx0",pmb->ncells1);
+  ParArrayND<Real> dx1("dx1",pmb->ncells1);
+
+  // this is obviously overkill for this constant velocity problem
+  for (int k=ks; k<=ke; k++) {
+    for (int j=js; j<=je; j++) {
+      pmb->pcoord->CenterWidth1(k, j, is, ie, dx0);
+      pmb->pcoord->CenterWidth2(k, j, is, ie, dx1);
+      for (int i=is; i<=ie; i++) {
+        min_dt = std::min(min_dt, dx0(i)/std::abs(vx));
+        min_dt = std::min(min_dt, dx1(i)/std::abs(vy));
+      }
+    }
+  }
+
+  return cfl*min_dt;
+}
+
+// Compute fluxes at faces given the constant velocity field and 
+// some field "advected" that we are pushing around.
+// This routine implements all the "physics" in this example
+TaskStatus CalculateFluxes(Container<Real>& rc) {
+  MeshBlock *pmb = rc.pmy_block;
+  int is = pmb->is; int js = pmb->js; int ks = pmb->ks;
+  int ie = pmb->ie; int je = pmb->je; int ke = pmb->ke;
+
+  CellVariable<Real>& q = rc.Get("advected");
+  auto pkg = pmb->packages["Advection"];
+  const auto& vx  = pkg->Param<Real>("vx");
+  const auto& vy  = pkg->Param<Real>("vy");
+
+  int maxdim = std::max(std::max(pmb->ncells1, pmb->ncells2), pmb->ncells3);
+  ParArrayND<Real> ql("ql",maxdim);
+  ParArrayND<Real> qr("qr",maxdim);
+  ParArrayND<Real> qltemp("qltemp",maxdim);
+
+  // get x-fluxes
+  for (int k = ks; k <= ke; k++) {
+    for (int j = js; j <= je; j++) {
+      // get reconstructed state on faces
+      pmb->precon->DonorCellX1(k, j, is-1, ie+1, q.data, ql, qr);
+      if (vx > 0.0) {
+        for (int i=is; i<=ie+1; i++) {
+          q.flux[0](k,j,i) = ql(i)*vx;
+        }
+      } else {
+        for (int i=is; i<=ie+1; i++) {
+          q.flux[0](k,j,i) = qr(i)*vx;
+        }
+      }
+    }
+  }
+  // get y-fluxes
+  if (pmb->pmy_mesh->ndim>=2) {
+    for (int k=ks; k<=ke; k++) {
+      pmb->precon->DonorCellX2(k, js-1, is, ie, q.data, ql, qr);
+      for (int j=js; j<=je+1; j++) {
+        pmb->precon->DonorCellX2(k, j, is, ie, q.data, qltemp, qr);
+        if (vy > 0.0) {
+          for (int i=is; i<=ie; i++) {
+            q.flux[1](k,j,i) = ql(i)*vy;
+          }
+        } else {
+          for (int i=is; i<=ie; i++) {
+            q.flux[1](k,j,i) = qr(i)*vy;
+          }
+        }
+        auto temp = ql;
+        ql = qltemp;
+        qltemp = temp;
+      }
+    }
+  }
+
+  // TODO(jcd): implement z-fluxes
+
+  return TaskStatus::success;
+
+}
+
+} // namespace Advection
+
+// *************************************************//
+// define the application driver. in this case,    *//
+// that just means defining the MakeTaskList       *//
+// function.                                       *//
+// *************************************************//
+// first some helper tasks
+TaskStatus UpdateContainer(MeshBlock *pmb, int stage, std::vector<std::string>& stage_name, Integrator* integrator) {
```

Comment:
Since in this case, updating a container is integration, I wonder if this method could have a more descriptive name, like `Substep` or something.